src/main/java/org/jabref/logic/util/io/FileUtil.java

```diff
@@ -118,6 +132,103 @@ public static Optional<String> getUniquePathFragment(List<String> paths, Path da
                               .map(part -> part.substring(0, part.lastIndexOf(File.separator)));
     }
 
+    public static Path getAppDataBackupDir() {
+        Path directory = Path.of(AppDirsFactory.getInstance().getUserDataDir(
+                                     "jabref",
+                                     new BuildInfo().version.toString(),
+                                     "org.jabref"))
+                             .resolve("backups");
+        return directory;
+    }
+
+    /**
+     * Determines the path of the backup file (using the given extension)
+     *
+     * <p>
+     *     As default, a directory inside the user temporary dir is used.<br>
+     *     In case a AUTOSAVE backup is requested, a timestamp is added
+     * </p>
+     * <p>
+     *     <em>SIDE EFFECT</em>: Creates the directory.
+     *     In case that fails, the return path of the .bak file is set to be next to the .bib file
+     * </p>
+     * <p>
+     *     Note that this backup is different from the <code>.sav</code> file generated by {@link org.jabref.logic.autosaveandbackup.BackupManager}
+     *     (and configured in the preferences as "make backups")
+     * </p>
+     */
+    public static Path getPathForNewBackupFileAndCreateDirectory(Path targetFile, BackupFileType fileType) {
```

Comment:
The (original) idea of `FileUtil` is to add general functionality that could possibly be even in the `nio` java package, such as getting/adding file extension etc. These additions here look like they are particular to the backup and should thus be moved to a backup-specific class (backupmanager?)

Reply:
More than 50% of these functionalities will also be used in #8750. I can craft out these methods in another utility class to enable reuse and to keep the (semantic) coupling of these methods.